File: Leetcode/389. Find the Difference/find_the_difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        ans =0
        for i in s:
            ans ^=ord(i)
        for j in t:
            ans ^=ord(j)
        return chr(ans) 
        
    def findTheDifference1(self, s: str, t: str) -> str:
        s_sum,t_sum = 0,0
        for i in s:
            s_sum+=ord(i)
        for j in t:
            t_sum+=ord(j)
        return chr(t_sum-s_sum) 
    

sol = Solution()
print(sol.findTheDifference("abcd","abcde"))
print(sol.findTheDifference1("abcd","abcde"))

File: Leetcode/389. Find the Difference/find_the_difference.rs
struct Solution{}

impl Solution {
    pub fn find_the_difference(s: String, t: String) -> char {
        let mut ans =0;
        for i in s.chars(){
            ans ^= i as u8;
        }
        for j in t.chars(){
            ans ^= j as u8;
        }
        return char::from(ans)
    }
    pub fn find_the_difference1(s: String, t: String) -> char {
        let mut s_sum=0;
        let mut t_sum=0;
        for i in s.chars(){
            s_sum += i as u16;
        }
        for j in t.chars(){
            t_sum += j as u16;
        }
        return char::from((t_sum-s_sum) as u8)
    }
}
fn main() {
    let res = Solution::find_the_difference(String::from("abcd"),String::from("abcde"));
    println!("{}",res);
    let res1 = Solution::find_the_difference1(String::from("abcd"),String::from("abcde"));
    println!("{}",res1)
}
